sotodlib/core/metadata/obsdb.py

```diff
@@ -92,39 +92,63 @@ def __init__(self, map_file=None, init_db=True, wafer_info=None):
                     changes = True
             if changes:
                 self.conn.commit()
-        self.primary_keys = self.get_primary_fields()
+        self.primary_keys = self.get_primary_fields(wafer_info)
 
-    def get_primary_fields(self):
+    def get_primary_fields(self, wafer_info=None):
```

Comment:
I guess this should be a private function.  Seems like you cache `self.primary_keys` on init, so that's what users should access.